dataset/dataset.go

```diff
@@ -68,6 +68,18 @@ func NewAPIClient(datasetAPIURL string) *Client {
 	}
 }
 
+// NewAPIClientWithMaxRetries creates a new instance of Client with a given dataset api url and the relevant tokens,
+// setting a number of max retires for the HTTP client
+func NewAPIClientWithMaxRetries(datasetAPIURL string, maxRetries int) *Client {
```

Comment:
I assume dimension extractor needed a different retry count to the default?

Also remember `/health` and `/healthcheck` endpoints are whitelisted to only ever retry once